### PR TITLE
feat(frontend): first rough implementation of the claims page

### PR DESCRIPTION
**Pull request by luzzif**

### CODE REVIEW

File: packages/frontend/src/components/claims/chain-overview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        abi: metromAbi,
+        address:
+            CHAIN_DATA[chainWithClaimsData.chain.id as SupportedChain].contract
+                .address,
+        functionName: "claimRewards",
+        args: [
+            !account
+                ? []
+                : chainWithClaimsData.claims.map((claim) => {
+                      console.log(claim.amount);
```
Comment (guerrap) [inline packages/frontend/src/components/claims/chain-overview/index.tsx:55]: Console log here.

File: packages/frontend/src/components/claims/chain-overview/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            } catch (error) {
+                console.warn("could not create kpi token", error);
+            } finally {
+                setClaiming(false);
+            }
+        };
+        void create();
+    }, [publicClient, simulatedClaimAll, writeContractAsync]);
+
+    console.log({ error });
```
Comment (guerrap) [inline packages/frontend/src/components/claims/chain-overview/index.tsx:99]: Same here.

File: packages/frontend/src/components/claims/chains/styles.module.css [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.root {
+    @apply w-72 flex flex-col gap-2 bg-white rounded-xl p-4;
+}
+
+.header {
+    @apply mb-4;
+}
+
+.row {
+    @apply flex p-3 gap-3 rounded-lg hover:bg-gray-100 cursor-pointer;
```
Comment (guerrap) [inline packages/frontend/src/components/claims/chains/styles.module.css:10]: The usual duration/transitions classes where we change properties on hover.

File: packages/frontend/src/components/claims/index.tsx [hunk 4]
```diff
@@ -1,12 +1,76 @@
...
 export function Claims() {
     const { loading, claims } = useClaims();
 
-    // TODO: remove this
-    console.log({ loading, claims });
+    const chainsWithClaimsData = useMemo(() => {
+        if (loading || claims.length === 0) return [];
+        const reduced = claims.reduce(
+            (acc, claim) => {
+                let data = acc[claim.chainId as SupportedChain];
```
Comment (guerrap) [inline packages/frontend/src/components/claims/index.tsx:26]: `data` can be a const here, no?

File: packages/frontend/src/components/claims/chain-overview/index.tsx (hunk 1 above)
Comment (luzzif) [reply packages/frontend/src/components/claims/chain-overview/index.tsx]: Addressed in fc2090d0f245c01946f822083b8adbd743cd8f78

File: packages/frontend/src/components/claims/chain-overview/index.tsx (hunk 2 above)
Comment (luzzif) [reply packages/frontend/src/components/claims/chain-overview/index.tsx]: Addressed in fc2090d0f245c01946f822083b8adbd743cd8f78

File: packages/frontend/src/components/claims/chains/styles.module.css (hunk 3 above)
Comment (luzzif) [reply packages/frontend/src/components/claims/chains/styles.module.css]: Addressed in fc2090d0f245c01946f822083b8adbd743cd8f78

File: packages/frontend/src/components/claims/index.tsx (hunk 4 above)
Comment (luzzif) [reply packages/frontend/src/components/claims/index.tsx]: Addressed in fc2090d0f245c01946f822083b8adbd743cd8f78

**Review by guerrap — APPROVED** [review]

**Pull request merged**